hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -176,19 +202,72 @@ public void incrementCount(List<Long> txIDs)
    */
   @Override
   public int resetCount(List<Long> txIDs) throws IOException {
-    lock.lock();
+    final int batchSize = 100;
+    int totalProcessed = 0;
+    long startTxId = 0;
+
     try {
+      // If txIDs are not provided, fetch all failed transactions in batches
       if (txIDs == null || txIDs.isEmpty()) {
-        txIDs = getFailedTransactions(LIST_ALL_FAILED_TRANSACTIONS, 0).stream()
-            .map(DeletedBlocksTransaction::getTxID)
-            .collect(Collectors.toList());
+        List<DeletedBlocksTransaction> batch;
+        do {
+          // Fetch the batch of failed transactions
+          batch = getFailedTransactionsBatch(batchSize, startTxId);
+
+          // If the batch is empty, skip further processing
+          if (!batch.isEmpty()) {
+            List<Long> batchTxIDs = batch.stream()
+                .map(DeletedBlocksTransaction::getTxID)
+                .collect(Collectors.toList());
+
+            lock.lock();
+            try {
+              transactionStatusManager.resetRetryCount(batchTxIDs);
+              int batchProcessed = deletedBlockLogStateManager.resetRetryCountOfTransactionInDB(
+                  new ArrayList<>(batchTxIDs));
+
+              totalProcessed += batchProcessed;
+            } finally {
+              lock.unlock();
+            }
+
+            // Update startTxId to continue from the last processed transaction in the next iteration
+            startTxId = batch.get(batch.size() - 1).getTxID() + 1;
+          }
+
+        } while (!batch.isEmpty());
+      } else {
+        // Process txIDs provided by the user in batches
```

Comment:
The user provided list of `txIDs` reaches SCM via RPC call, so it's ok to process this in single go.

Reply:
Done